test/unit/question_base_test.rb

```diff
@@ -90,6 +90,19 @@ class QuestionBaseTest < ActiveSupport::TestCase
         assert_equal [:done], @question.permitted_next_nodes
       end
     end
+
+    context 'next_node called with block but not permitted option specified' do
+      should 'return nodes returned via syntactic sugar methods' do
+        @question.next_node do |response|
+          if response == 'yes'
+            outcome :done
+          else
+            question :another_question
+          end
+        end
+        assert_equal [:done, :another_question], @question.permitted_next_nodes
+      end
+    end
   end
```

Comment:
Regarding your first comment - that was a typo - well spotted! Fixup commit: 30d92cccbbbfcb8d3952414a26b76a85d8d3a3c6.